### PR TITLE
OS: Added initial support for AIX

### PR DESCRIPTION
**Pull request by konimex**
### Features

This adds support for AIX systems, most of the functions have worked as intended.

### Issues

We'll have to rely on external source (our current primary source is #591) since we don't have access to any AIX systems, be it from a VM, or a physical one.

### Working state

- [x] get_model
- [x] get_distro
- [x] get_kernel
- [x] get_uptime
- [x] get_packages
- [x] get_cpu
- [x] get_cpu_usage
- [x] get_gpu
- [x] get_memory
- [x] get_resolution
- [x] get_de
- [x] get_wm
- [x] get_disk
- [x] <s>get_battery</s>
- [x] get_local_ip
- [x] <s>get_install_date</s>
- [x] get_term

### CODE REVIEW

**Comment by dylanaraps** [comment]
AIX runs CDE by default with KDE and GNOME also install-able. \[1\] 

You can also run an X server on AIX. \[2\]

I'm not sure if all of the display stuff uses X or not. We may need to ask @bobblestiltskin.

\[1\] https://en.wikipedia.org/wiki/IBM_AIX#Graphical 

\[2\] https://www.ibm.com/support/knowledgecenter/ssw_aix_71/com.ibm.aix.aixwnpgd/doc/aixwnpgd/winman.htm

**Comment by dylanaraps** [comment]
We can leave `get_kernel` out since it doesn't display anything useful on AIX.

https://en.wikipedia.org/wiki/Uname (Scroll to the AIX entries in the table)

**Comment by dylanaraps** [comment]
This might be helpful:

www.unixmantra.com/2013/09/aix-listing-cpu-memory-ram-harddisks-ip-address.html

**Comment by dylanaraps** [comment]
I found a website where we can buy access to an AIX machine for $20 a day. \[1\] I'll buy a days worth and give you access, just let me know what day/time is best for you. If we need another day or two, that's fine, I'll just order more days.

We can also pick AIX version, should we go with the latest? (7.1 TL4)

\[1\] https://www.siteox.com/cart.php

File: neofetch [hunk 1]
```diff
@@ -274,6 +279,10 @@ get_model() {
         "Solaris")
             model="$(prtconf -b | awk -F':' '/banner-name/ {printf $2}')"
         ;;
+
+        "AIX")
+            model="$(prtconf | awk -F':' '/System Model/ {printf $2}')"
```
Comment (dylanaraps) [inline neofetch]: Maybe we can merge this and Solaris?

`awk '/banner-name|System Model/ {printf $2}'`

Comment (konimex) [inline neofetch]: Solaris uses `prtconf -b`, it won't work with AIX.

Comment (dylanaraps) [inline neofetch]: I see, all good.

**Comment by konimex** [comment]
>I found a website where we can buy access to an AIX machine for $20 a day. [1] I'll buy a days worth and give you access, just let me know what day/time is best for you. If we need another day or two, that's fine, I'll just order more days.

>We can also pick AIX version, should we go with the latest? (7.1 TL4)

>[1] https://www.siteox.com/cart.php

Actually, I'd like to hold out at least until everything except `get_battery` and `get_install_date` completed.

We'll confirm with @bobblestiltskin if everything works fine then I'll do some finishing.

File: neofetch [hunk 2]
```diff
@@ -1189,6 +1223,12 @@ get_memory() {
             mem_used="$(sysinfo -mem | awk -F '\\/|)' '{print $2; exit}')"
             mem_used="$((${mem_used/max} / 1024 / 1024))"
         ;;
+
+        "AIX")
+            mem_stat=($(svmon -G -O unit=MB))
+            mem_total=${mem_stat[7]}
```
Comment (dylanaraps) [inline neofetch]: These need to be quoted. `mem_total="${mem_stat[7]}"`

**Comment by dylanaraps** [comment]
I made an ascii logo.

```sh
"AIX"*)
    set_colors 2 7
    ascii_file="aix"
;;
```

```sh
${c1}           `:+ssssossossss+-`           
        .oys///oyhddddhyo///sy+.        
      /yo:+hNNNNNNNNNNNNNNNNh+:oy/      
    :h/:yNNNNNNNNNNNNNNNNNNNNNNy-+h:    
  `ys.yNNNNNNNNNNNNNNNNNNNNNNNNNNy.ys   
 `h+-mNNNNNNNNNNNNNNNNNNNNNNNNNNNNm-oh  
 h+-NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN.oy 
/d`mNNNNNNN/::mNNNd::m+:/dNNNo::dNNNd`m:
h//NNNNNNN: . .NNNh  mNo  od. -dNNNNN:+y
N.sNNNNNN+ -N/ -NNh  mNNd.   sNNNNNNNo-m
N.sNNNNNs  +oo  /Nh  mNNs` ` /mNNNNNNo-m
h//NNNNh  ossss` +h  md- .hm/ `sNNNNN:+y
:d`mNNN+/yNNNNNd//y//h//oNNNNy//sNNNd`m-
 yo-NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNm.ss 
 `h+-mNNNNNNNNNNNNNNNNNNNNNNNNNNNNm-oy  
   sy.yNNNNNNNNNNNNNNNNNNNNNNNNNNs.yo   
    :h+-yNNNNNNNNNNNNNNNNNNNNNNs-oh-    
      :ys:/yNNNNNNNNNNNNNNNmy/:sy:      
        .+ys///osyhhhhys+///sy+.        
            -/osssossossso/-            
```

Source for logo: https://en.wikipedia.org/wiki/IBM_AIX

File: neofetch [hunk 3]
```diff
@@ -2538,6 +2579,10 @@ get_distro_colors() {
     #
     # $ascii_distro is the same as $distro.
     case "$ascii_distro" in
+        "AIX"*)
+            set_colors 2 7
+            ascii_file="aix"
+        ;;
```
Comment (dylanaraps) [inline neofetch]: Add a line break here. :P

File: neofetch [hunk 4]
```diff
@@ -304,6 +304,9 @@ get_title() {
 }
 
 get_kernel() {
+    # Since AIX is an integrated system, it's better to skip this function altogether
+    [[ $os == "AIX" ]] && return
```
Comment (dylanaraps) [inline neofetch]: Can you quote `$os`?

**Comment by konimex** [comment]
So, apart from memory issues (that have been fixed since), all core functions seem to work perfectly. I can merge this now, or wait until I figure out `get_packages` (`lslpp` doesn't seem to produce any useful outputs) and `get_install_date` (their `ls` seem doesn't have any equivalent to `--full-time`).

**Comment by dylanaraps** [comment]
Just add `AIX` to the OS case in `get_install_date` and add something like this:

```sh
"AIX") err "Install Date doesn't work on AIX due to the 'ls' missing --full-time"; return ;;
```

**Comment by dylanaraps** [comment]
Once you've added the above, I'm happy for this to be merged. :+1:

**Comment by bobblestiltskin** [comment]
On Mon, Jan 09, 2017 at 05:02:39AM -0800, Herdiansyah wrote:
> So, apart from memory issues (that have been fixed since), all core functions seem to work perfectly. I can merge this now, or wait until I figure out `get_packages` (`lslpp` doesn't seem to produce any useful outputs) and `get_install_date` (their `ls` seem doesn't have any equivalent to `--full-time`).
> 

Hello

You could use lslpp as follows to get all of the packages, if you
want the count?

bash-4.4$ lslpp -q -l -J | grep -v ^#
  DirectorCommonAgent
  DirectorPlatformAgent
  GSKit8.gskcrypt64.ppc.rte
  GSKit8.gskssl64.ppc.rte
  ICU4C.rte
  Java5.sdk
  Java5_64.sdk
  Java6.sdk
  Tivoli_Management_Agent.client.rte
  X11.adt.bitmaps
  X11.adt.imake
  X11.adt.include
  X11.adt.lib
  X11.apps.aixterm

....

bash-4.4$ lslpp -q -l -J | grep -v ^# | wc -l
    1092
bash-4.4$ 

where ...

       -J
            Generates output in a form suitable for the System
Management Interface Tool (SMIT) command to list output. This flag
can only be specified with the -l (lowercase L) and -L flags.
       -l
            (lowercase L) Displays the name, most recent level,
state, and description of the specified fileset.
       -q
            Suppresses the display of column headings.

Bob

**Comment by konimex** [comment]
>You could use lslpp as follows to get all of the packages, if you want the count?

Now this is what I'm talking about. Thanks!

Anyway, does `lslpp` also cover RPM packages? If not we'll have to add `rpm -qa | wc -l` outputs.

**Comment by bobblestiltskin** [comment]
On Mon, Jan 09, 2017 at 06:08:55AM -0800, Herdiansyah wrote:
> >You could use lslpp as follows to get all of the packages, if you want the count?
> 
> Now this is what I'm talking about. Thanks!
> 
> Anyway, does `lslpp` also cover RPM packages? If not we'll have to add `rpm -qa | wc -l` outputs.

No, lslpp does not count rpm packages. It seems ...

# rpm -qa | grep screen
screen-4.0.3-1
# lslpp -J -q -l | grep -v ^# | grep screen
# 

So, I concur that the best single number for number of packages
will be the sum of the counts derived from lslpp and rpm -qa.

Bob

**Comment by konimex** [comment]
Alright, I'll merge this into the master branch tomorrow morning if there's no problem.

**Comment by bobblestiltskin** [comment]
On Mon, Jan 09, 2017 at 07:14:32AM -0800, Herdiansyah wrote:
> Alright, I'll merge this into the master branch tomorrow morning if there's no problem.

Looks good to me, thank you very much :-)

**Pull request merged**